File: sdk/python/examples/examples_utils.py
from binascii import unhexlify
from pathlib import Path

from symbolchain.CryptoTypes import PrivateKey
from symbolchain.PrivateKeyStorage import PrivateKeyStorage
from symbolchain.symbol.KeyPair import KeyPair


def read_contents(filepath) -> str:
	with open(filepath, 'rt', encoding='utf8') as infile:
		return infile.read()


def read_private_key(filepath, password=None) -> KeyPair:
	if not isinstance(filepath, Path):
		filepath = Path(filepath)

	if filepath.suffix == '.txt':
		return KeyPair(PrivateKey(unhexlify(read_contents(filepath).strip())))

	storage = PrivateKeyStorage(filepath.parent, password)
	return KeyPair(storage.load(filepath.stem))

File: sdk/javascript/examples/examples_utils.js
import { PrivateKey } from '../src/index.js';
import { KeyPair } from '../src/symbol/index.js';
import fs from 'fs';

export const readContents = filepath => fs.readFileSync(filepath, { encoding: 'utf8', flag: 'r' });
export const readPrivateKey = filepath => new KeyPair(new PrivateKey(readContents(filepath).trim()));
